tutorial/10_key_features/004_distributed.py

```diff
@@ -4,88 +4,329 @@
+            href="#multi-thread-optimization",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>InMemoryStorage</U></TD></TR>
+                    <TR><TD><U>JournalStorage</U></TD></TR>
+                </TABLE>
+            >
+        ];
+
+        singlenode_storages [
+            shape=box,
+            style=rounded,
+            href="#multi-process-optimization",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>JournalStorage</U></TD></TR>
+                    <TR><TD><U>RDBStorage</U></TD></TR>
+                </TABLE>
+            >
+        ]
+
+        rdb_storage [
+            shape=box,
+            style=rounded,
+            href="#multi-node-optimization",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>RDBStorage</U></TD></TR>
+                </TABLE>
+            >
+        ]
+
+        grpc_storage [
+            shape=box,
+            style=rounded,
+            href="#grpc-storage-proxy",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>GrpcStorageProxy</U></TD></TR>
+                </TABLE>
+            >
+        ]
+
+        multithread -> multithread_storages [label="Multi-thread"];
+        multithread -> single_node [label="Multi-process"];
+        single_node -> singlenode_storages [label="Single node"];
+        single_node -> many_nodes [label="Multi-node"];
+        many_nodes -> rdb_storage [label="No"];
+        many_nodes -> grpc_storage [label="Yes"];
+    }
+
+.. _multi-thread-optimization:
+
+Multi-thread Optimization
+-------------------------
+
+.. Note::
+    **Recommended backends**:
+        - :class:`~optuna.storages.InMemoryStorage`
+        - :class:`~optuna.storages.JournalStorage`
+        - :class:`~optuna.storages.RDBStorage`
+
+
+You can run multiple trials in parallel just by setting the ``n_jobs``
+parameter in :meth:`~optuna.study.Study.optimize()`.
+
+Multi-thread optimization has traditionally been inefficient in Python due to the Global Interpreter Lock (GIL).
+However, starting from Python 3.14 (pending official release), the GIL is expected to be removed.
+This change will make multi-threading a good option, especially for parallel optimization.
 
 """
+
+# %%
+
+import optuna
+from optuna.storages import JournalStorage
+from optuna.storages.journal import JournalFileBackend
+from optuna.trial import Trial
+import threading
+
+
+def objective(trial: Trial):
+    print(f"Running trial {trial.number=} in {threading.current_thread().name}")
+    x = trial.suggest_float("x", -10, 10)
+    return (x - 2) ** 2
+
+
+study = optuna.create_study(
+    study_name="journal_storage_multithread",
+    storage=JournalStorage(JournalFileBackend(file_path="./journal.log")),
+    load_if_exists=True,
+)
```

Comment:
```suggestion
study = optuna.create_study(
    storage=JournalStorage(JournalFileBackend(file_path="./journal.log")),
)
```
I was wondering whether it’s necessary to specify `study_name` and set `load_if_exists=True` when running optimization with multiple threads. If it's not strictly necessary, I thought it might be better to keep the example minimal.